vdr/didx509/jwt_utils.go

```diff
@@ -0,0 +1,27 @@
+package didx509
+
+import (
+	"context"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"strings"
+)
+
+// ExtractProtectedHeaders extracts the protected headers from a JWT string.
+// The function takes a JWT string as input and returns a map of the protected headers.
+// Note that:
+//   - This method ignores strings that don't look like JWTs and returns an empty map.
+//   - This method ignores any parsing errors and returns an empty map instead of an error.
+func ExtractProtectedHeaders(jwt string) (map[string]interface{}, error) {
```

Comment:
inverting the conditions / removing all nested statements makes this a lot easier to read / maintain

Reply:
The conditions are removed, let the parse method figure it out.